arviz/data/utils.py

```diff
@@ -136,3 +143,50 @@ def extract(
     if num_samples is not None:
         data = data.isel(sample=slice(None, num_samples))
     return data
+
+
+def drop_objects_from_dataset(data: Dataset) -> Tuple[Dataset, List[str], List[str]]:
+    """Returns a new ``Dataset`` without object variables and coords."""
+    vars = {}
+    vars_dropped = []
+    for k, v in data.data_vars.items():
+        if v.dtype.hasobject:
+            vars_dropped.append(k)
+            continue
+        vars[k] = v
+
+    coords = {}
+    coords_dropped = []
+    for k, v in data.coords.items():
+        if v.dtype.hasobject:
+            coords_dropped.append(k)
+            continue
+        coords[k] = v
+
+    ndata = Dataset(
+        data_vars=vars,
```

Comment:
Will this copy all data?

Reply:
The `Dataset` has just three kwargs and I'm passing all of them.

Reply:
I think the original question was about copying the data unnecessarly and wasting memory instead of more data than desired missing from the resulting dataset 🤔. If that were the case, it won't copy the data, xarray stores the arrays in datasets without copying them